osu.Game.Rulesets.Osu/Skinning/Default/DefaultSpinnerDisc.cs

```diff
@@ -137,6 +137,8 @@ private void updateStateTransforms(DrawableHitObject drawableHitObject, ArmedSta
                 this.ScaleTo(initial_scale);
                 this.RotateTo(0);
 
+                updateComplete(false, 0);
```

Comment:
Can this nor be moved into an `else` block alongside the other line?

Reply:
No, because the spinner accent fades from white to blue during preempt, and it needs to be re-added so that during rewind it still is blue. TimeCompleted isn't reset to null if you rewind to before the spinner completes, it retains its value. So if the line is moved to the else block it won't actually execute on rewinds and the spinner won't have its blue accent (unless you rewind to before the spinner appears).

Though now that I think about it, I just realized this means that the gold fade will be stacked twice if you rewind to before spinner completion. It'll be played once because the fade gets queued by the `BeginAbsoluteSequence` block, but it'll also be played by the `complete.BindValueChanged`.

Reply:
I would hope for the bindable feedback to be removed entirely, but that would mean figuring out a way to trigger `updateStateTransforms` once `TimeCompleted != null`... maybe this is fine for now?

Alternatively you could change the bindable feedback to trigger `updateStateTransforms` rather than `updateDiscColour` so that there is only one path for updating disc colours, but not sure there's much of a gain there.